packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -81,9 +88,22 @@ export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
         }
     }
 
-    public shouldRunLastSummary(): boolean {
+    public shouldRunLastSummary(stopReason: SummarizerStopReason): boolean {
+        if (stopReason !== "parentNotConnected") {
```

Comment:
@vladsud Thinking about this case a bit more, does it even make sense to have the logic at this level? For example, we have the `allowLastSummary` parameter on the `RunningSummarizer.waitStop(...)` method. But our value for `allowLastSummary` may not matter because of this stop reason check. I think it makes more sense for the caller of `RunningSummarizer.waitStop(...)` to make this decision on a specific stop reason not allowing summarization.

Reply:
Sure, we can move decision up the stack.